app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/Filter.kt

```diff
@@ -0,0 +1,231 @@
+                        text = option.description,
+                        onClick = { parkingViewModel.toggleProtection(option) },
+                        value = selectedProtection.contains(option),
+                        testTag = "ProtectionFilterItem")
+                  }
+                }
+          }
+
+      // Rack type filter
+      FilterSection(
+          title = stringResource(R.string.list_screen_rack_type),
+          expandedState = showRackTypeOptions,
+          expandable = displayHeader,
+          onReset = { parkingViewModel.clearRackType() },
+          onApply = { parkingViewModel.selectAllRackTypes() }) {
+            LazyRow(
+                modifier = Modifier.fillMaxWidth().testTag("RackTypeFilter"),
+                horizontalArrangement = Arrangement.spacedBy(4.dp)) {
+                  items(ParkingRackType.entries.toTypedArray()) { option ->
+                    ToggleButton(
+                        text = option.description,
+                        onClick = { parkingViewModel.toggleRackType(option) },
+                        value = selectedRackTypes.contains(option),
+                        testTag = "RackTypeFilterItem")
+                  }
+                }
+          }
+
+      // Capacity filter
+      FilterSection(
+          title = stringResource(R.string.list_screen_capacity),
+          expandedState = showCapacityOptions,
+          expandable = displayHeader,
+          onReset = { parkingViewModel.clearCapacity() },
+          onApply = { parkingViewModel.selectAllCapacities() }) {
+            LazyRow(
+                modifier = Modifier.fillMaxWidth().testTag("CapacityFilter"),
+                horizontalArrangement = Arrangement.spacedBy(4.dp)) {
+                  items(ParkingCapacity.entries.toTypedArray()) { option ->
+                    ToggleButton(
+                        text = option.description,
+                        onClick = { parkingViewModel.toggleCapacity(option) },
+                        value = selectedCapacities.contains(option),
+                        testTag = "CapacityFilterItem")
+                  }
+                }
+          }
+
+      // CCTV filter with checkbox
+      Row(
+          modifier = Modifier.fillMaxWidth(),
+          verticalAlignment = Alignment.CenterVertically,
+          horizontalArrangement = Arrangement.spacedBy(4.dp)) {
+            Checkbox(
+                checked = onlyWithCCTV,
+                onCheckedChange = { parkingViewModel.setOnlyWithCCTV(it) },
+                modifier = Modifier.testTag("CCTVCheckbox"),
+                colors = getCheckBoxColors(ColorLevel.PRIMARY))
+            Text(
+                text = stringResource(R.string.list_screen_display_only_cctv),
+                style = MaterialTheme.typography.bodyMedium,
+                textAlign = TextAlign.Start,
+                modifier = Modifier.testTag("CCTVCheckboxLabel"))
+          }
+    }
+  }
+}
+
+@Composable
+fun FilterSection(
```

Comment:
To be honest i would add documentation here and for all new composable functions that are not screens. Now it is not blocking for me since we didn't do that before. If you choose not to address it, we musn't forget to mention it for the polishing during the meeting.

Reply:
No I think you are right. It is relevant here. It also is for the `FilterHeader.` I am also quite annoyed by the name `FilterHeader` because it's not just the header, it's the whole thing. I don't know if you see what I mean. Should I change it?

Reply:
Honestly for me the name is good since I directly understood with regard to its implementation for list screen. I would change it for `FilterLayout` which is more general but not that different so you can keep it as is.

Reply:
I decided to change it because it was really misleading to me. I changed it to `FilterPanel` which is way more accurate. Now this panel has a header (displayed if displayHeader is true) and an expandable content. Much clearer

Reply:
That's perfect and the documention is complete thanks !